validated/dash_apps/finished_apps/daily_validation.py

```diff
@@ -84,15 +145,205 @@
+        html.Div(
+            id="daily-status-message",
+            children=[""],
+            style={"font-family": DEFAULT_FONT},
+        ),
         html.H1(
             children="Detail of Selected Day",
             style={"font-family": DEFAULT_FONT},
         ),
         table_detail,
+        html.Button("Add row", id="detail-add-button"),
+        html.Button("Save to Validated", id="detail-save-button"),
+        html.Button("Reset Validated", id="detail-reset-button"),
+        html.Div(
+            id="detail-status-message",
+            children=[""],
+            style={"font-family": DEFAULT_FONT},
+        ),
         html.H1(
             children="Plot",
             style={"font-family": DEFAULT_FONT},
         ),
-        dcc.Graph(figure=plot),
+        dcc.Graph(id="plot", figure=plot),
     ]
 )
+
+
+@app.callback(
+    [
+        Output("daily-status-message", "children"),
+        Output("detail-status-message", "children"),
+        Output("plot", "figure"),
+        Output("table_daily", "rowData"),
+        Output("table_detail", "rowData"),
+        Output("table_detail", "rowTransaction"),
+        Output("table_detail", "scrollTo"),
+    ],
+    [
+        Input("daily-save-button", "n_clicks"),
+        Input("daily-reset-button", "n_clicks"),
+        Input("detail-save-button", "n_clicks"),
+        Input("detail-reset-button", "n_clicks"),
+        Input("detail-add-button", "n_clicks"),
+    ],
+    [
+        State("table_daily", "selectedRows"),
+        State("table_detail", "selectedRows"),
+        State("table_detail", "rowData"),
+    ],
+    prevent_initial_call=True,
+)
+def buttons_callback(
+    daily_save_clicks: int,
+    daily_reset_clicks: int,
+    detail_save_clicks: int,
+    detail_reset_clicks: int,
+    detail_add_clicks: int,
+    in_daily_selected_rows: list[dict],
+    in_detail_selected_rows: list[dict],
+    in_detail_row_data: list[dict],
+) -> tuple[str, str, go.Figure, list[dict], list[dict], list[dict], list[dict]]:
+    """Callback for buttons adding and resetting Validated data
+
+    Args:
+        daily_save_clicks (int): Number of times daily-save-button was clicked
+        daily_reset_clicks (int): Number of times daily-reset-button was clicked
+        detail_save_clicks (int): Number of times detail-save-button was clicked
+        detail_reset_clicks (int): Number of times detail-reset-button was clicked
+        detail_add_clicks (int): Number of times detail-add-button was clicked
+        in_daily_selected_rows (list[dict]): Selected rows in table_daily
+        in_detail_selected_rows (list[dict]): Selected rows in table_detail
+        in_detail_row_data (list[dict]): Full row data for table_detail
+
+    Returns:
+        tuple[str, str, go.Figure, list[dict], list[dict], dict, dict]:
+            Callback outputs
+    """
+    global DATA_DAILY, DATA_DETAIL
+
+    ctx = dash.callback_context
+    if not ctx.triggered:
+        return (dash.no_update,) * 7
```

Comment:
This was just because callbacks are usually triggered when the app first loads so this would prevent anything happening in that case. However, I also have `prevent_initial_call=True` so this is redundant and I will remove it.